Directory.Packages.props

```diff
@@ -28,6 +28,7 @@
     <PackageVersion Include="System.IO.Pipelines" Version="7.0.0" />
     <PackageVersion Include="System.IO.Pipes" Version="4.3.0" />
     <PackageVersion Include="System.Net.Http" Version="4.3.4" />
+    <PackageVersion Include="System.Text.Json" Version="7.0.2" />
```

Comment:
Does both .net fx and .net core compilations remove an assembly from output if it is not used in the main assembly? I remember seeing that behavior before but couldn't be sure.

Reply:
No, I've never heard of that feature. The build cannot be confident that a dll isn't used, even if the compiler doesn't need it. If you wanted to suppress deployment of an assembly, you can add `ExcludeAssets="runtime"` to a PackageReference.
But this doesn't add a package reference. It merely specifies the version to use when some project _does_ reference the package.

Reply:
I think I was remembering an issue I encountered for vstest/mstest scenario in VS repo when a referenced but unused assembly wasn't included in test out directory.

In any case, this does get referenced in StreamJsonRpc though so my question was more around for those components using StreamJsonRpc but are not interested in System.Text.Json support. Can they add a package reference to System.Text.Json on their own and exclude assets to save space on their final packaging as long as they dont use SystemTextJsonFormatter.

Reply:
Yes, they can do that.